proposals/NNNN-sdl-protocol-security-specification.md

```diff
@@ -0,0 +1,394 @@
+# SDL Protocol Security Specification
+
+* Proposal: [SDL-NNNN](NNNN-sdl-protocol-security-specification.md)
+* Author: [Kujtim Shala](https://github.com/smartdevicelink/kshala-ford)
+* Status: **Awaiting review**
+* Impacted Platforms: [Core / iOS / Java Suite / JavaScript Suite / Protocol]
+
+## Introduction
+
+This proposal adds the documentation and specification around protected communication, encryption and handshake to the SDL protocol specification. It is a collaborated effort with Ford and Luxoft to protect the implemented SDL security from unexpected changes and to allow evolutionary proposals to improve it.
+
+## Motivation
+
+SDL uses protected services since the start of mobile navigation apps. Unfortunately the protocol specification doesn't contain a section around establishing a protected service with handshake and encryption. This leads to contradictory implementations in SDL Core and the libraries. Recently we have identified a number of bugs in SDL Core and the libraries which break with protected communication. These issues are difficult to address with the SDLC as there is no specification around, that Core and libraries should follow. In order to resolve the issues there must be a proper specification within SDLC around protected communication.
```

Comment:
```suggestion
SDL has used protected services since the start of mobile navigation apps. Unfortunately the protocol specification doesn't contain a section around establishing a protected service with handshake and encryption. This leads to contradictory implementations in SDL Core and the app libraries. Recently we have identified a number of bugs in SDL Core and the app libraries which break with protected communication. These issues are difficult to address with the SDLC as there is no specification around protected communication that Core and the app libraries should follow. In order to resolve the issues there must be a proper specification within SDLC around protected communication.
```